libc/test/UnitTest/HermeticTestUtils.cpp

```diff
@@ -104,6 +104,8 @@ void *__dso_handle = nullptr;
 
 } // extern "C"
 
+void *operator new(unsigned long size, void *ptr) { return ptr; }
```

Comment:
is this okay or should it call realloc?

Reply:
This is variation (9) in the list: https://en.cppreference.com/w/cpp/memory/new/operator_new
and it is said that: "The standard library implementation performs no action and returns `ptr` unmodified."